### PR TITLE
Add plot method to all DataModules

### PR DESCRIPTION
**Pull request by adamjstewart**
Now that (almost) all of our Datasets have a plot method, our DataModules should too. This PR adds a plot method to all DataModules except ChesapeakeCVPR, which is still missing a plot method. See #253.

### CODE REVIEW

File: torchgeo/datamodules/oscd.py [hunk 1]
```diff
@@ -196,3 +197,10 @@ def test_dataloader(self) -> DataLoader[Any]:
         return DataLoader(
             self.test_dataset, batch_size=1, num_workers=self.num_workers, shuffle=False
         )
+
+    def plot(self, *args: Any, **kwargs: Any) -> plt.Figure:
+        """Run :meth:`torchgeo.datasets.OSCD.plot`.
+
+        .. versionadded:: 0.4
+        """
+        return self.test_dataset.plot(*args, **kwargs)
```
Comment (calebrob6) [inline torchgeo/datamodules/oscd.py:206]: Heads up that this won't work because the sample that comes in here will be a flattened image

Comment (adamjstewart) [reply torchgeo/datamodules/oscd.py:206]: Yep, I noticed a few failures like this. Not sure what the best approach is other than to try to make things more uniform (dataset.__getitem__ should always return C x H x W, plot method should always convert to H x W x C). Prob going to take me a while to get these working uniformly. If anyone else wants to hack on this PR go ahead.

Comment (calebrob6) [reply torchgeo/datamodules/oscd.py:206]: Well the issue is that Dataset .plot() methods expects a sample as returned by __getitem__, whereas the samples that are used within the train/val methods in our trainers have been modified by the DataModule preprocessing functions

Comment (calebrob6) [reply torchgeo/datamodules/oscd.py:206]: E.g. in OSCD, `sample["image"]` as returned by `__getitem__` will be something like (2 x C x H x W) whereas in the trainer it will be converted to (2C x H x W). If we do channel-wise standardization, this will also be applied during the train/val steps.

Comment (adamjstewart) [reply torchgeo/datamodules/oscd.py:206]: In that case, we should either change the dataset to return 2C x H x W, or create a new ObjectDetectionTask that can handle 2 x C x H x W samples without any changes.

Comment (calebrob6) [reply torchgeo/datamodules/oscd.py:206]: Do you mean `ChangeDetectionTask`?

Generally, what happens if the DataModule subtracts the mean and divides by the stdev?

Comment (adamjstewart) [reply torchgeo/datamodules/oscd.py:206]: Yes.

Not really sure how to handle this. Kornia transforms have a reverse operation, right? Maybe we could apply that if it exists.

Comment (isaaccorley) [reply torchgeo/datamodules/oscd.py:206]: Kornia has a denormalize but unfortunately it doesn't work per channel it only takes in a single value for mean and std.

Here's a torchvision denormalize I've been using lately:

```

class Denormalize(nn.Sequential):
    def __init__(
        self, mean: Sequence[float] = IMAGENET_MEAN, std: Sequence[float] = IMAGENET_STD
    ) -> None:
        self.mean = mean
        self.std = std
        self.inv_mean = torch.tensor([-x for x in mean])
        self.inv_std = torch.tensor([1 / x for x in std])
        super().__init__(
            T.Normalize(mean=[0.0, 0.0, 0.0], std=self.inv_std),
            T.Normalize(mean=self.inv_mean, std=[1.0, 1.0, 1.0]),
        )
```

Comment (calebrob6) [reply torchgeo/datamodules/oscd.py:206]: So I think we should factor out augmentations and preprocessing from the datamodules (where we can easily support denormalizing in a consistent way)

**Comment by adamjstewart** [comment]
Okay, I backed out a few datamodule plot methods that were broken. Those can be added one at a time in future PRs by anyone who cares. I'm still hoping to add plot methods to most datamodules, even if normalization may make visualization a bit broken. I'm not really sure what to do about those other than undo datamodule-specific normalizations inside datamodule plot methods before passing to dataset plot methods.

**Review by ashnair1 — APPROVED** [review]

**Review by calebrob6 — APPROVED** [review]
